### PR TITLE
Introduce the --no-random-temp-subdir option

### PR DESCRIPTION
**Pull request by pmderodat**

### CODE REVIEW

File: src/e3/testsuite/__init__.py [hunk 1]
```diff
@@ -188,17 +188,26 @@ def testsuite_main(self, args: Optional[List[str]] = None) -> int:
         temp_group.add_argument(
             "-t", "--temp-dir", metavar="DIR", default=Env().tmp_dir
         )
+        temp_group.add_argument(
+            "--no-random-temp-subdir",
+            action="store_true",
```
Comment (brobecke) [inline src/e3/testsuite/__init__.py:193]: You might find it easier to read the code by avoiding a negative in the variable name. As we see below, you then declare a local variable called "random_subdir = not args.no_random_[...]". Instead, what you could do is use argparse "dest" parameter, default, and then set the action to "store_false". A bit more work here, but I makes the use easier, IMO.

File: src/e3/testsuite/__init__.py [hunk 2]
```diff
@@ -188,17 +188,26 @@ def testsuite_main(self, args: Optional[List[str]] = None) -> int:
...
             nargs="?",
             default=None,
             const="tmp",
-            help="Unlike --temp-dir, use this very directory to store"
-            " testsuite temporaries (i.e. no random subdirectory). Also"
-            " automatically disable temp dir cleanup, to be developer"
-            " friendly. If no directory is provided, use the local"
-            ' "tmp" directory',
+            help="Convenience shortcut for dev setups: forces `-t DIR"
+            " --no-random-temp-subdir --cleanup-mode=none` and cleanup `DIR`"
```
Comment (brobecke) [inline src/e3/testsuite/__init__.py:208]: "cleanup" -> "cleans up" (that's the only way I could understand this part of the sentence).

File: tests/tests/test_basics.py [hunk 3]
```diff
@@ -305,7 +307,44 @@ def test(self, caplog):
...
+    def test(self, tmp_path):
+        run_testsuite(
+            self.Mysuite,
+            [
+                f"--temp-dir={tmp_path}",
+                "--no-random-temp-subdir",
+                "--cleanup-mode=none",
+            ],
+        )
+        assert os.path.isdir(str(tmp_path / "test1"))
```
Comment (brobecke) [inline tests/tests/test_basics.py:346]: Out of curiosity, what type is tmp_path, and how does this work?

**Review by brobecke — COMMENTED** [review]
Thanks very much, Pierre-Marie. Overall, this look very good, and I wouldn't have a problem if this were approved and pushed as is.

I did notice a couple of very minor things, and while at it, I saw something I had a question on...

File: src/e3/testsuite/__init__.py (hunk 1 above)
Comment (pmderodat) [reply src/e3/testsuite/__init__.py:193]: Good idea, thanks!

File: src/e3/testsuite/__init__.py (hunk 2 above)
Comment (pmderodat) [reply src/e3/testsuite/__init__.py:208]: Right, fixed

File: tests/tests/test_basics.py (hunk 3 above)
Comment (pmderodat) [reply tests/tests/test_basics.py:346]: See the corresponding doc in the py.test framework: https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmp-path-fixture

**Review by brobecke — APPROVED** [review]
Looks great. Thanks PM!

**Comment by pmderodat** [comment]
Sure, and thank you for the review!

**Pull request merged**